iceoryx_hoofs/include/iceoryx_hoofs/cxx/convert.hpp

```diff
@@ -81,6 +82,14 @@ class convert
     template <typename Destination>
     static bool fromString(const char* v, Destination& dest) noexcept;
 
+    /// @brief Sets dest from a given string. If the conversion fails false is
+    ///         returned and the value of dest is undefined.
+    /// @param[in] v string which contains the value of dest
+    /// @param[in] dest destination to which the value should be written
+    /// @return false = if the conversion fails otherwise true
+    template <uint64_t Capacity>
+    static bool fromString(const char* v, string<Capacity>& dest) noexcept;
```

Comment:
Is this tested? Maybe indirectly by passing a `string<Capacity>` to the previously existing template but then why do we need this overload in the first place?

If it is not tested we should add at least three tests, one with an empty string, one with a non-empty one and one with maximum length `Capacity`.